swagger/v1/api_spec.json

```diff
@@ -2784,8 +2784,58 @@
           }
         }
       },
-      "ParticipantDeclarationDataRequest": {
-        "description": "A participant declaration data request",
+      "ParticipantDeclarationPost2024ECTDataRequest": {
+        "description": "A participant declaration data request for ECT participants in cohort 2025 and later years",
+        "type": "object",
+        "properties": {
+          "type": {
+            "type": "string",
+            "enum": [
+              "participant-declaration"
+            ]
+          },
+          "attributes": {
+            "anyOf": [
+              {
+                "$ref": "#/components/schemas/ECFParticipantDeclarationStartedAttributesRequest"
+              },
+              {
+                "$ref": "#/components/schemas/ECFParticipantDeclarationRetainedAttributesRequest"
+              },
+              {
+                "$ref": "#/components/schemas/ECFParticipantDeclarationCompletedAttributesRequest"
+              },
+              {
+                "$ref": "#/components/schemas/ECFParticipantDeclarationExtendedAttributesRequest"
+              }
+            ]
+          }
+        }
+      },
+      "ParticipantDeclarationPost2024MentorDataRequest": {
+        "description": "A participant declaration data request for mentor participants in cohort 2025 and later years",
+        "type": "object",
+        "properties": {
+          "type": {
+            "type": "string",
+            "enum": [
+              "participant-declaration"
+            ]
+          },
+          "attributes": {
+            "anyOf": [
+              {
+                "$ref": "#/components/schemas/ECFParticipantDeclarationStartedAttributesRequest"
```

Comment:
For mentors, do we need to differentiate the parameters as well? ie: `course_identifier` can be either `ecf-induction` or `ecf-mentor` or it should be `ecf-mentor` only?

Reply:
Good question; this would lead to doubling the number of attribute request schemas, so I'd say not if we can help it, but I'll defer to @cwrw 🙂

Reply:
I think let's leave it as is, although good point to avoid duplication